src/lowercase.rs

```diff
@@ -44,12 +51,33 @@ impl<'a> Lowercase<'a> {
         &self.0
     }
 
-    pub fn into_owned(self) -> Lowercase<'static> {
-        Lowercase(Cow::Owned(self.0.into_owned()))
+    pub fn into_cow(self) -> Cow<'a, str> {
+        self.0
     }
 
     pub fn to_uppercase(&self) -> String {
-        self.as_str().to_uppercase()
+        self.as_str().to_ascii_uppercase()
+    }
+
+    /// Like [`str::strip_prefix`]. Takes a prefix that is known to be already lowercase.
+    pub fn strip_prefix_unchecked<S: Borrow<str>>(&'a self, prefix: S) -> Option<Lowercase<'a>> {
+        self.0.strip_prefix(prefix.borrow()).map(|s| Self(Cow::Borrowed(s)))
+    }
+
+    /// Like [`str::strip_suffix`]. Takes a suffix that is known to be already lowercase.
+    pub fn strip_suffix_unchecked<S: Borrow<str>>(&'a self, suffix: S) -> Option<Lowercase<'a>> {
+        self.0.strip_suffix(suffix.borrow()).map(|s| Self(Cow::Borrowed(s)))
+    }
+
+    #[cfg(any(feature = "vic3", feature = "imperator"))]
+    pub fn rmatch_indices_unchecked(&self, separator: char) -> RMatchIndices<char> {
+        self.0.rmatch_indices(separator)
+    }
+
+    /// Implementing `SliceIndex` directly requires some nightly-only API
+    #[cfg(any(feature = "vic3", feature = "imperator"))]
+    pub fn slice<R: 'a + SliceIndex<str, Output = str>>(&'a self, range: R) -> Self {
```

Comment:
you can implement [std::ops::Index](https://doc.rust-lang.org/std/ops/trait.Index.html)

Reply:
Aha I didn't know that worked on ranges

Reply:
I tried but I can't get it to work because the trait method returns a reference. I can construct a `Lowercase` to return, but I can't return a reference to one I construct inside the function.